contracts/BrandedToken.sol

```diff
@@ -190,6 +190,19 @@ contract BrandedToken is Organized, EIP20Token {
         valueToken = _valueToken;
         conversionRate = _conversionRate;
         conversionRateDecimals = _conversionRateDecimals;
+
+        bytes memory data = abi.encodeWithSignature("decimals()");
+        (, bytes memory returnData) = address(_valueToken).call(data);
```

Comment:
> Question:
> Since _valueToken is an external contract, a malicious 🐛 user can pass _valueToken contract which has implemented decimals() function with malicious code. e.g. Run a loop which consumes all the gas.
> 
> Should we have check that msg.sender should not be equal to ValueToken contract to prevent re-entrancy 🤔 ? Thinking loud here do you think it's a concern 🔴 ?


This will be called while deployment, msg.sender will be the deployer. The params in the constructor are sent by the deployer itself. Deployer has full control here, the deployer can choose any address and pay the gas.

Reply:
@abhayks1: dovetailing with @deepesh-kn's view, I am generally not in favor of building in limits on the consumer.